### PR TITLE
Ports: No more `config.sub` patches.

### PR DESCRIPTION
**Pull request by timschumi**
This removes all the (semi-)manual `config.sub` patching that has been going on, and additionally eliminates a few usages of `autoconf` and friends along the way.

### CODE REVIEW

File: Ports/glu/package.sh [hunk 1]
```diff
@@ -2,29 +2,14 @@
...
-    run cp config.sub config.sub.contents
-    run mv -f config.sub.contents config.sub
-    run sed -i 's/-haiku/-serenity/' config.sub
-}
-
-post_configure() {
-    unset ACLOCAL
-    unset GL_CFLAGS
-    unset GL_LIBS
+    run sed -i 's/haiku/serenity/' config.sub
```
Comment (linusg) [inline Ports/glu/package.sh]: :grin:

**Review by linusg — COMMENTED** [review]
>  Ports: Replace config_{sub,guess}_path with a proper array 

needs an explanation please

File: Ports/glu/package.sh (hunk 1 above)
Comment (timschumi) [reply Ports/glu/package.sh]: Justice for Haiku!

(There is no actual requested change here, right?)

Comment (linusg) [reply Ports/glu/package.sh]: nope, all good, just think it's funny :)

**Pull request merged**